pkgs/stdenv/adapters.nix

```diff
@@ -108,6 +108,20 @@ rec {
     });
   });
 
+  # Puts all the other ones together
+  makeStatic = stdenv: lib.foldl (lib.flip lib.id) stdenv (
+    lib.optional stdenv.hostPlatform.isDarwin makeStaticDarwin
+
+    ++ [ makeStaticLibraries propagateBuildInputs ]
+
+    # Apple does not provide a static version of libSystem or crt0.o
+    # So we can’t build static binaries without extensive hacks.
+    ++ lib.optional (!stdenv.hostPlatform.isDarwin) makeStaticBinaries
+
+    # Glibc doesn’t come with static runtimes by default.
+    # ++ lib.optional (stdenv.hostPlatform.libc == "glibc") ((lib.flip overrideInStdenv) [ self.stdenv.glibc.static ])
```

Comment:
Does this line have a future?

It has been unchanged since its introduction 3 years ago https://github.com/NixOS/nixpkgs/commit/3c36dccd7ca4bcc3a3899e794864780dc7a4ce8c but this move breaks the little context we have. This might be a good time to delete this commented code.

Reply:
fwiw `makeStaticBinaries` already has some special casing for this (by adding `glibc.static` to `extraTargetLibraries`)

Reply:
I'll delete it after --- I think the context was always just the thing I moved it with.

Reply:
@sternenseemann since that one had the typo, neither is doing anything. Maybe we just would compile glibc properly now? Or maybe no one is testing without musl?

Reply:
I remember using `makeStaticBinaries` and it working, but my suspicion is that in such cases using anything from `pkgsStatic` will then leak musl in automatically, so in effect it may very well be that nobody is testing without musl.